File: check_tags.py
import re

with open('frontend/pages/admin/slots/[id].vue', 'r', encoding='utf-8') as f:
    text = f.read()

template_match = re.search(r'<template>(.*?)</template>', text, re.DOTALL)
if template_match:
    template_content = template_match.group(1)
    lines = template_content.split('\n')

    stack = []

    for i, line in enumerate(lines):
        for m in re.finditer(r'<(/)?(div|form|nav|section|main|header|footer)\b[^>]*?(/)?>', line, re.IGNORECASE):
            if bool(m.group(3)): continue
            tag = m.group(2).lower()
            if bool(m.group(1)):
                if stack and stack[-1][0] == tag:
                    stack.pop()
                else:
                    print(f"Mismatch at line {i+1}: expected </{stack[-1][0] if stack else 'none'}> but found </{tag}>. Stack size: {len(stack)}")
                    if stack:
                        print(f"Top was opened at line {stack[-1][1]}")
            else:
                stack.append((tag, i+1))

    print("Remaining in stack:", [(t[0], t[1]) for t in stack])

File: check_tags.js
const fs = require('fs');
const content = fs.readFileSync('frontend/pages/admin/slots/[id].vue', 'utf8');
const template = content.split('<template>')[1].split('</template>')[0];
const lines = template.split('\n');
const tags = [];
const regex = /<\/?(div|form)[^>]*>/gi;

let match;
while ((match = regex.exec(template)) !== null) {
    const tagStr = match[0];
    if (tagStr.endsWith('/>')) continue;
    const tagName = match[1].toLowerCase();
    const isClosing = tagStr.startsWith('</');

    // Calculate line number
    const prefix = template.substring(0, match.index);
    const lineNum = prefix.split('\n').length + 1; // +1 because <template> is line 1 usually, though let's just say relative to template content

    if (isClosing) {
        if (tags.length && tags[tags.length - 1].tag === tagName) {
            tags.pop();
        } else {
            console.log(`Mismatch at line ${lineNum}! Found ${tagStr} but expected ${tags.length ? tags[tags.length - 1].tag : 'none'} (Opened at line ${tags.length ? tags[tags.length - 1].line : '?'})`);
            break;
        }
    } else {
        tags.push({ tag: tagName, line: lineNum, content: tagStr.replace(/\s+/g, ' ').substring(0, 50) });
    }
}
